main/service/external/external_exchange_api_service.go

```diff
@@ -0,0 +1,54 @@
+package external
+
+import (
+	"container/list"
+	"errors"
+
+	"github.com/sirupsen/logrus"
+	"ses.genesis.com/exchange-web-service/main/config"
+
+	"github.com/go-resty/resty/v2"
+)
+
+type RateAPI interface {
+	GetRate(from, to string) (float64, error)
+}
+
+type Service struct {
+	config       *config.AppConfig
+	client       *resty.Client
+	externalAPIs *list.List
+}
+
+func NewExternalExchangeAPIService(conf *config.AppConfig, client *resty.Client, apis *list.List) *Service {
+	return &Service{
+		config:       conf,
+		client:       client,
+		externalAPIs: apis,
+	}
+}
+
+func (controller *Service) CurrentRate(from, to string) (float64, error) {
+	return getRate(controller.externalAPIs.Front(), from, to)
+}
+
+func getRate(val *list.Element, from, to string) (float64, error) {
```

Comment:
```suggestion
func (s *Service) getRate(val *list.Element, from, to string) (float64, error) {
```

Оскільки getRate використовується лише в рамках цього сервісу, є сенс зробити його методом